waiter/src/waiter/core.clj

```diff
@@ -1067,6 +1067,14 @@
    :refresh-service-descriptions-fn (pc/fnk [[:state kv-store]]
                                       (fn refresh-service-descriptions-fn [service-ids]
                                         (sd/refresh-service-descriptions kv-store service-ids)))
+   :request->consent-service-id (pc/fnk [[:state kv-store service-description-builder service-id-prefix waiter-hostnames]
```

Comment:
Why are we defining a separate function for the *consent* service-id? Shouldn't we just have one consistent way to compute the service-id? It seems like maintaining separate functions is setting us up for more inconsistency in the future.

Reply:
This gap is reduced in the new way, as now we use the common `compute-descriptor` with an override for allowing run-as-requester without permission (the consent workflow is supposed to give us the permission) to get the service ID.